service/service.go

```diff
@@ -211,9 +211,10 @@ func (srv *Service) initExtensionsAndPipeline(ctx context.Context, set Settings,
 		return fmt.Errorf("cannot build pipelines: %w", err)
 	}
 
-	if cfg.Telemetry.Metrics.Level != configtelemetry.LevelNone && cfg.Telemetry.Metrics.Address != "" {
-		// The process telemetry initialization requires the ballast size, which is available after the extensions are initialized.
-		if err = proctelemetry.RegisterProcessMetrics(srv.telemetryInitializer.ocRegistry, getBallastSize(srv.host)); err != nil {
+
+	if set.Config.Service.Telemetry.Metrics.Level != configtelemetry.LevelNone && set.Config.Service.Telemetry.Metrics.Address != "" {
+		processMetrics := proctelemetry.NewProcessMetrics(srv.telemetrySettings.Logger, srv.telemetryInitializer.registry, srv.telemetryInitializer.mp, getBallastSize(srv.host))
+		if err = processMetrics.RegisterProcessMetrics(context.Background(), srv.telemetryInitializer.ocRegistry); err != nil {
```

Comment:
I think it would be better if we could keep a similar function definition here and just pass down the feature gate registry + meterprovider, something like, no need to make this build a `newProcessMetrics`, just do it inside the `RegisterProcessMetrics`:
```
if err = proctelemetry.RegisterProcessMetrics(srv.telemetryInitializer.ocRegistry, getBallastSize(srv.host), meterProvider, registry); err != nil {
```

Reply:
Got it I will take your suggestion in that case - it makes sense to minimize the amount of unnecessary changes i make in this pr.  I just really dislike having a function with too many arguments and it seemed like it was simple enough to avoid this. 

My thought process was to split the arguments into two groups. (1) arguments that will be fields of processMetric struct and can therefore be passed into `NewProcessMetrics()` and (2) arguments that are needed for metric recording that can be passed into `RegisterProcessMetrics()` i.e. the context and the ocRegistry. It also felt easier to test because creation of the processMetric struct could be done separately